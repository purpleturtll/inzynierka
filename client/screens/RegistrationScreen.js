import React, { useState } from 'react';
import {
  View, ScrollView, Text, TextInput, StyleSheet, TouchableOpacity
} from 'react-native';
import { Feather } from '@expo/vector-icons';

const RegistrationScreen = ({ navigation }) => {

  const marginLeftText = '5%';
  const marginBottomText = 5;

  const [data, setData] = useState({
    firstname: '',
    surname: '',
    email: '',
    password: '',
    check_textInputChange: false,
    passwordEye: true,
    passwordConfirmationEye: true
  });

  const handleFirstnameChange = (val) => {
    setData({
      ...data,
      firstname: val
    });
  }

  const handleSurnameChange = (val) => {
    setData({
      ...data,
      surname: val
    });
  }

  const handleEmailChange = (val) => {
    setData({
      ...data,
      email: val
    });
  }

  const handlePasswordChange = (val) => {
    setData({
      ...data,
      password: val
    });
  }

  const updatePasswordEye = () => {
    setData({
      ...data,
      passwordEye: !data.passwordEye
    })
  }

  const updatePasswordConfirmationEye = () => {
    setData({
      ...data,
      passwordConfirmationEye: !data.passwordConfirmationEye
    })
  }

  const onRegisterPress = () => {

    const res = fetch('http://10.0.2.2:8080/auth/register', {
      body: JSON.stringify({
        firstname: data.firstname,
        surname: data.surname,
        email: data.email,
        password: data.password
      }),
      headers: {"Content-Type": "application/json"},
      method:'POST'
    }).then((response)=>{
      if(response.status==201){
        navigation.navigate('DoneRegistrationScreen')
      }
      else{
        navigation.navigate('RegistrationScreen')
      }
    })
  }

  return (
    <ScrollView>
      <View style={styles.container}>
        <Text style={styles.title}>Rejestracja</Text>
        <Text style={{ marginLeft: marginLeftText, marginBottom: marginBottomText }}>Imię</Text>
        <TextInput
          placeholderTextColor="#000"
          placeholderStyle={{}}
          style={styles.textInput}
          onChangeText={(val) => handleFirstnameChange(val)}
        />

        <View>
          <Text style={{ marginLeft: marginLeftText, marginBottom: marginBottomText }}>Nazwisko</Text>
          <TextInput
            placeholderTextColor="#000"
            placeholderStyle={{}}
            style={styles.textInput}
            onChangeText={(val) => handleSurnameChange(val)}
          />
        </View>

        <View>
          <Text style={{ marginLeft: marginLeftText, marginBottom: marginBottomText }}>Adres e-mail</Text>
          <TextInput
            placeholderTextColor="#000"
            placeholderStyle={{}}
            style={styles.textInput}
            autoCapitalize="none"
            onChangeText={(val) => handleEmailChange(val)}
          />
        </View>
        <View>
          <Text style={{ marginLeft: marginLeftText, marginBottom: marginBottomText }}>Hasło</Text>
          <View style={styles.passwordContainer}>
            <TextInput
              placeholderTextColor="#000"
              secureTextEntry={data.passwordEye ? true : false}
              style={styles.textInput, styles.textInputPass}
              autoCapitalize="none"
              onChangeText={(val) => handlePasswordChange(val)}
            />
            <TouchableOpacity
              onPress={updatePasswordEye}>
              {data.passwordEye ?
                <Feather
                  name="eye-off"
                  color="grey"
                  size={20}
                  style={{ marginTop: 3 }}
                />
                :
                <Feather
                  name="eye"
                  color="grey"
                  size={20}
                  style={{ marginTop: 3 }}
                />
              }
            </TouchableOpacity>
          </View>
        </View>
        <View>
          <Text style={{ marginLeft: marginLeftText, marginBottom: marginBottomText }}>Potwierdź hasło</Text>
          <View style={styles.passwordContainer}>
            <TextInput
              placeholderTextColor="#000"
              secureTextEntry={data.passwordConfirmationEye ? true : false}
              style={styles.textInput, styles.textInputPass}
              autoCapitalize="none"
              onChangeText={(val) => handlePasswordChange(val)}
            />
            <TouchableOpacity
              onPress={updatePasswordConfirmationEye}>
              {data.passwordConfirmationEye ?
                <Feather
                  name="eye-off"
                  color="grey"
                  size={20}
                  style={{ marginTop: 3 }}
                />
                :
                <Feather
                  name="eye"
                  color="grey"
                  size={20}
                  style={{ marginTop: 3 }}
                />
              }
            </TouchableOpacity>
          </View>
        </View>
        <View style={styles.buttonContainer}>
          <TouchableOpacity style={styles.registerButton} onPress={onRegisterPress}>
            <Text style={{ color: '#fff' }}>Utwórz konto</Text>
          </TouchableOpacity>
          <TouchableOpacity onPress={() => navigation.navigate('SignInScreen')}>
            <Text style={{ color: '#69667C', fontWeight: 'bold', marginTop: 10 }}>Masz już konto?</Text>
          </TouchableOpacity>
        </View>
      </View>
    </ScrollView>
  )
}
export default RegistrationScreen;

const styles = StyleSheet.create({
  container: {
    width: '85%',
    marginLeft: '7.5%'
  },
  title: {
    fontSize: 30,
    fontWeight: '900',
    textAlign: 'center',
    marginTop: 15,
    marginBottom: 20
  },
  textInput: {
    width: '100%',
    paddingHorizontal: 20,
    paddingVertical: 10,
    backgroundColor: '#E2E1E1',
    borderRadius: 40,
    marginBottom: 15,
  },
  passwordContainer: {
    flexDirection: 'row',
    width: '100%',
    paddingHorizontal: 20,
    paddingVertical: 10,
    backgroundColor: '#E2E1E1',
    borderRadius: 40,
    marginBottom: 15
  },
  textInputPass: {
    flex: 1,
  },
  buttonContainer: {
    alignItems: 'center',
    marginTop: 0
  },
  registerButton: {
    flexDirection: 'row',
    height: 50,
    width: '50%',
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#362893',
    borderRadius: 20,
    padding: 10,
  },
});
