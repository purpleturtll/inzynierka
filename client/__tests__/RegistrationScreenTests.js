import React from 'react';
import {create} from 'react-test-renderer';
import RegistrationScreen from '../screens/RegistrationScreen';

import { shallow } from "enzyme";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
configure({ adapter: new Adapter() });

const tree = create(<RegistrationScreen/>);

test('snapshot', () => {
  expect(tree).toMatchSnapshot();
});


if('should change state if email is entered', () => {
  const instanceOf = renderer.create(<RegistrationScreen/>).getInstance();
  instanceOf.handleEmailChange('abcd@gmail.com');
  expect(instanceOf.state.email).toEqual('abcd@gmail.com');
});


