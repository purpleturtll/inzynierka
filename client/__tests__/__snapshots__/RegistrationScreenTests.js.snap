// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<RCTScrollView>
  <View>
    <View
      style={
        Object {
          "marginLeft": "7.5%",
          "width": "85%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 33,
            "fontWeight": "bold",
            "marginBottom": 10,
            "marginTop": 20,
            "textAlign": "center",
          }
        }
      >
        Rejestracja
      </Text>
      <Text
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "900",
            "marginBottom": 5,
            "marginLeft": "5%",
            "marginTop": 10,
          }
        }
      >
        Imię
      </Text>
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholderStyle={Object {}}
        placeholderTextColor="#000"
        rejectResponderTermination={true}
        style={
          Array [
            Object {
              "backgroundColor": "#E2E1E1",
              "borderRadius": 40,
              "marginBottom": 10,
              "paddingHorizontal": 20,
              "paddingVertical": 10,
              "width": "100%",
            },
            null,
          ]
        }
        underlineColorAndroid="transparent"
      />
      <View>
        <Text
          style={
            Object {
              "fontSize": 16,
              "fontWeight": "900",
              "marginBottom": 5,
              "marginLeft": "5%",
              "marginTop": 10,
            }
          }
        >
          Nazwisko
        </Text>
        <TextInput
          allowFontScaling={true}
          onChangeText={[Function]}
          placeholderStyle={Object {}}
          placeholderTextColor="#000"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "backgroundColor": "#E2E1E1",
                "borderRadius": 40,
                "marginBottom": 10,
                "paddingHorizontal": 20,
                "paddingVertical": 10,
                "width": "100%",
              },
              null,
            ]
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View>
        <Text
          style={
            Object {
              "fontSize": 16,
              "fontWeight": "900",
              "marginBottom": 5,
              "marginLeft": "5%",
              "marginTop": 10,
            }
          }
        >
          Adres e-mail
        </Text>
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          onChangeText={[Function]}
          placeholderStyle={Object {}}
          placeholderTextColor="#000"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "backgroundColor": "#E2E1E1",
                "borderRadius": 40,
                "marginBottom": 10,
                "paddingHorizontal": 20,
                "paddingVertical": 10,
                "width": "100%",
              },
              null,
            ]
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <View>
        <Text
          style={
            Object {
              "fontSize": 16,
              "fontWeight": "900",
              "marginBottom": 5,
              "marginLeft": "5%",
              "marginTop": 10,
            }
          }
        >
          Hasło
        </Text>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#E2E1E1",
                "borderRadius": 40,
                "flexDirection": "row",
                "marginBottom": 15,
                "paddingHorizontal": 20,
                "paddingVertical": 10,
                "width": "100%",
              },
              null,
            ]
          }
        >
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            onChangeText={[Function]}
            placeholderTextColor="#000"
            rejectResponderTermination={true}
            secureTextEntry={true}
            style={
              Object {
                "flex": 1,
              }
            }
            underlineColorAndroid="transparent"
          />
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text />
          </View>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 15,
                "marginBottom": 10,
                "marginLeft": "5%",
              },
              null,
            ]
          }
        >
          min. 8 znaków • wielka litera • mała litera • cyfra • znak specjalny
        </Text>
      </View>
      <View>
        <Text
          style={
            Object {
              "fontSize": 16,
              "fontWeight": "900",
              "marginBottom": 5,
              "marginLeft": "5%",
              "marginTop": 10,
            }
          }
        >
          Potwierdź hasło
        </Text>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#E2E1E1",
                "borderRadius": 40,
                "flexDirection": "row",
                "marginBottom": 15,
                "paddingHorizontal": 20,
                "paddingVertical": 10,
                "width": "100%",
              },
              null,
            ]
          }
        >
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            onChangeText={[Function]}
            placeholderTextColor="#000"
            rejectResponderTermination={true}
            secureTextEntry={true}
            style={
              Object {
                "flex": 1,
              }
            }
            underlineColorAndroid="transparent"
          />
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "marginBottom": 20,
            "marginTop": 10,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "#362893",
              "borderRadius": 20,
              "flexDirection": "row",
              "height": 50,
              "justifyContent": "center",
              "opacity": 1,
              "padding": 10,
              "width": "50%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#fff",
                "fontSize": 16,
              }
            }
          >
            Utwórz konto
          </Text>
        </View>
        <View
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#69667C",
                "fontSize": 15,
                "fontWeight": "bold",
                "marginTop": 10,
              }
            }
          >
            Masz już konto?
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTScrollView>
`;
